packages/story-editor/src/app/media/utils/useMediaInfo.js

```diff
@@ -0,0 +1,181 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useCallback, useMemo } from '@googleforcreators/react';
+import { getTimeTracker, trackError } from '@googleforcreators/tracking';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../config';
+import { MEDIA_VIDEO_DIMENSIONS_THRESHOLD } from '../../../constants';
+
+function loadScriptOnce(url) {
+  if (document.querySelector(`script[src="${url}"]`)) {
+    return Promise.resolve();
+  }
+
+  return new Promise((resolve, reject) => {
+    const script = document.createElement('script');
+    script.async = true;
+    script.crossOrigin = 'anonymous';
+    script.src = url;
+    script.addEventListener('load', resolve);
+    script.addEventListener('error', reject);
+    document.head.appendChild(script);
+  });
+}
+
+/**
+ * @typedef MediaInfo
+ * @property {string} mimeType File mime type.
+ * @property {number} fileSize File size in bytes.
+ * @property {string} format File format.
+ * @property {string} codec File codec.
+ * @property {number} frameRate Frame rate (rounded).
+ * @property {number} height Height in px.
+ * @property {number} width Width in px.
+ * @property {string} colorSpace Color space.
+ * @property {number} duration Video duration.
+ * @property {string} videoCodec Video codec.
+ * @property {string} audioCodec Audio codec.
+ * @property {boolean} isMuted Whether the video is muted.
+ */
+
+/**
+ * Custom hook to interact with mediainfo.js.
+ *
+ * @see https://mediainfo.js.org/
+ * @return {Object} Functions and vars related to mediainfo.js usage.
+ */
+function useMediaInfo() {
+  const { mediainfoUrl } = useConfig();
+
+  const getFileInfo = useCallback(
+    /**
+     * Returns information about a given media file.
+     *
+     * @param {File} file File object.
+     * @return {Promise<MediaInfo|null>} File info or null on error.
+     */
+    async (file) => {
+      const getSize = () => file.size;
+
+      const readChunk = (chunkSize, offset) =>
+        new Promise((resolve, reject) => {
+          const reader = new FileReader();
+          reader.onload = (event) => {
+            if (event.target.error) {
+              reject(event.target.error);
+            }
+            resolve(new Uint8Array(event.target.result));
+          };
+          reader.readAsArrayBuffer(file.slice(offset, offset + chunkSize));
+        });
+
+      //eslint-disable-next-line @wordpress/no-unused-vars-before-return -- False positive because of the finally().
+      const trackTiming = getTimeTracker('load_mediainfo');
+
+      try {
+        // This will expose the window.MediaInfo global.
+        await loadScriptOnce(mediainfoUrl);
+
+        const mediaInfo = await window.MediaInfo({ format: 'JSON' });
+        const result = JSON.parse(
+          await mediaInfo.analyzeData(getSize, readChunk)
+        );
+
```

Comment:
Could we use a service worker here?

Reply:
I have not looked into using a web worker at all, so this would require some exploration and performance testing first.

At first glance I doubt there's any benefit in using a web worker here though. If this is not an extensive call, then the whole overhead of using a web worker could even make it slower.